From 0ee012224447b6ec7f6f20d316b24449fdcc0d4b Mon Sep 17 00:00:00 2001
From: "Miss Islington (bot)"
 <31488909+miss-islington@users.noreply.github.com>
Date: Fri, 28 Feb 2025 09:22:15 +0100
Subject: [PATCH] [3.13] gh-129296: Fix `pyatomic.h` include paths (GH-129320)
 (#130667)

gh-129296: Fix `pyatomic.h` include paths (GH-129320)

Use relative includes in Include/cpython/pyatomic.h for
pyatomic_gcc.h, pyatomic_std.h and pyatomic_msc.h.

Do a similar change in Include/cpython/pythread.h for
pthread_stubs.h include.
(cherry picked from commit 3a974e39d54902699f360bc4db2fd351a6baf3ef)

Co-authored-by: Zanie Blue <contact@zanie.dev>
Upstream: 125ca028664ce72556e3983b57f521a118c4e677
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 Include/cpython/pyatomic.h | 6 +++---
 Include/cpython/pythread.h | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Include/cpython/pyatomic.h b/Include/cpython/pyatomic.h
index 4ecef4f56ed..28029859d3d 100644
--- a/Include/cpython/pyatomic.h
+++ b/Include/cpython/pyatomic.h
@@ -529,15 +529,15 @@ static inline void _Py_atomic_fence_release(void);
 
 #if _Py_USE_GCC_BUILTIN_ATOMICS
 #  define Py_ATOMIC_GCC_H
-#  include "cpython/pyatomic_gcc.h"
+#  include "pyatomic_gcc.h"
 #  undef Py_ATOMIC_GCC_H
 #elif __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
 #  define Py_ATOMIC_STD_H
-#  include "cpython/pyatomic_std.h"
+#  include "pyatomic_std.h"
 #  undef Py_ATOMIC_STD_H
 #elif defined(_MSC_VER)
 #  define Py_ATOMIC_MSC_H
-#  include "cpython/pyatomic_msc.h"
+#  include "pyatomic_msc.h"
 #  undef Py_ATOMIC_MSC_H
 #else
 #  error "no available pyatomic implementation for this platform/compiler"
diff --git a/Include/cpython/pythread.h b/Include/cpython/pythread.h
index 03f710a9f7e..e658b35bd90 100644
--- a/Include/cpython/pythread.h
+++ b/Include/cpython/pythread.h
@@ -22,7 +22,7 @@ PyAPI_DATA(const long long) PY_TIMEOUT_MAX;
     */
 #   define NATIVE_TSS_KEY_T     unsigned long
 #elif defined(HAVE_PTHREAD_STUBS)
-#   include "cpython/pthread_stubs.h"
+#   include "pthread_stubs.h"
 #   define NATIVE_TSS_KEY_T     pthread_key_t
 #else
 #   error "Require native threads. See https://bugs.python.org/issue31370"
-- 
2.48.1

